### ISSUE TITLE
When player changes to spectator he's always warned he can't switch for two minutes, even in single player or first two minutes of the round

### ISSUE THREAD

**Issue by handsm** [DarklightGames/DarkestHour#711]
With the new functionality to restrict frequent team switching, players can switch team freely in single player mode, or in the first two minutes of a round. When they can switch freely they are only asked to confirm the change and are not warned they won't be able to switch again for two minutes.

But when changing to be a spectator, they are always given this warning, even when they can switch freely and the warning isn't relevant and is misleading.

**Comment by handsm** [comment]
PLACEHOLDER - have fixed, will commit today.

**Comment by handsm** [comment]
Fixed in commit https://github.com/DarklightGames/DarkestHour/commit/9c1ee6782e0d6f4d803c3cd728ddbb2de6f50c50.

Re-factored the new change team functionality in the OnClick() function so the spectator switch is matched to the allied and axis functionality. All three options (allied, axis, spectator) are nearly identical, with only minor changes depending on the team being switched to, so I've grouped all three as one block in the switch statement.

OnClick() and it's supporting ChangeToXXXChoice() functions are now much shorter and with less repetition, as well as being standardised.
